actors/init/src/lib.rs

```diff
@@ -61,7 +61,19 @@ impl Actor {
 
         log::trace!("called exec; params.code_cid: {:?}", &params.code_cid);
 
-        let caller_code = rt.get_actor_code_cid(&rt.message().caller()).ok_or_else(|| {
+        let caller_addr = rt.message().caller();
+
+        let caller_id;
+        match caller_addr.id() {
+            Ok(id) => {
+                caller_id = id;
+            }
+            Err(e) => {
+                return Err(actor_error!(forbidden; "caller {} transfer to id address is not support, {}", caller_addr, e))
+            }
+        }
```

Comment:
Replace this with `rt.message().caller().id().unwrap()`. The syscall guarantees this will work. The same goes for `message().receiver()` and a number of other places where the runtime makes this promise.

(We might later consider `caller()` returning an ID, but that would be a larger change)